From e38e9639ab99a5da075ceaa86a9062f77a9b9df2 Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Thu, 14 Jun 2018 13:19:00 +0200
Subject: [PATCH 10/12] Add topic-miami devicetrees

Adds the devicetrees for topic system-on-modules to the kernel
---
 arch/arm/boot/dts/topic-miami-florida-mio.dts |   2 +
 .../arm/boot/dts/topic-miami-florida-mio.dtsi | 155 +++++++
 arch/arm/boot/dts/topic-miami-florida.dtsi    |  27 ++
 arch/arm/boot/dts/topic-miami.dts             |   3 +
 arch/arm/boot/dts/topic-miami.dtsi            | 331 +++++++++++++++
 arch/arm/boot/dts/topic-miamilite.dts         |   3 +
 arch/arm/boot/dts/topic-miamilite.dtsi        |  51 +++
 arch/arm/boot/dts/topic-miamiplus.dts         |   4 +
 arch/arm/boot/dts/topic-miamiplus.dtsi        |  97 +++++
 .../zynqmp-topic-miamimp-florida-gen.dts      | 282 +++++++++++++
 .../zynqmp-topic-miamimp-florida-test.dts     | 187 ++++++++
 .../boot/dts/xilinx/zynqmp-topic-miamimp.dts  | 377 +++++++++++++++++
 .../dts/xilinx/zynqmp-topic-miamiplusmp.dts   | 373 ++++++++++++++++
 arch/arm64/configs/topic_miamimp_defconfig    | 399 ++++++++++++++++++
 14 files changed, 2291 insertions(+)
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-mio.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-mio.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miami-florida.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miami.dts
 create mode 100644 arch/arm/boot/dts/topic-miami.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miamilite.dts
 create mode 100644 arch/arm/boot/dts/topic-miamilite.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miamiplus.dts
 create mode 100644 arch/arm/boot/dts/topic-miamiplus.dtsi
 create mode 100644 arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
 create mode 100644 arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-test.dts
 create mode 100644 arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts
 create mode 100644 arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
 create mode 100644 arch/arm64/configs/topic_miamimp_defconfig

diff --git a/arch/arm/boot/dts/topic-miami-florida-mio.dts b/arch/arm/boot/dts/topic-miami-florida-mio.dts
new file mode 100644
index 000000000000..5cf840f0ccfd
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-mio.dts
@@ -0,0 +1,2 @@
+/dts-v1/;
+/include/ "topic-miami-florida-mio.dtsi"
diff --git a/arch/arm/boot/dts/topic-miami-florida-mio.dtsi b/arch/arm/boot/dts/topic-miami-florida-mio.dtsi
new file mode 100644
index 000000000000..94b8df584280
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-mio.dtsi
@@ -0,0 +1,155 @@
+/include/ "topic-miami.dtsi"
+
+/* Common components for MIO parts of Florida boards. These are
+ * accessible even without programming any logic. */
+
+/ {
+	/* Regulator for the 5V USB VBUS power line. This is connected
+	 * to the second output of the GPIO expander. */
+	reg_usb0_vbus: regulator_usb {
+		compatible = "regulator-fixed";
+		regulator-name = "usb0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&gpioexf 1 0>;
+		enable-active-high;
+	};
+
+	/* USB PHY and reset IO pin */
+	usb_phy0: phy0 {
+		compatible = "usb-nop-xceiv";
+		#phy-cells = <0>;
+		reset-gpios = <&gpioex 0 1>;
+	};
+
+	xtal_27Mhz: clock_xtal_27Mhz {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <27000000>;
+	};
+
+	gpio_keys: gpio-keys {
+		compatible = "gpio-keys";
+		power {
+			label = "power";
+			gpios = <&gpioexf 2 1>; /* Input 2, active low */
+			linux,code = <116>; /* KEY_POWER */
+			gpio-key,wakeup;
+		};
+	};
+
+	gpio-poweroff {
+		compatible = "gpio-poweroff";
+		gpios = <&gpioexf 5 1>; /* Output 5, active low */
+		input;
+	};
+};
+
+/* I2C bus layout */
+&i2c0 {
+	i2cmux: i2cswitch@70 {
+		compatible = "nxp,pca9546";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0x70>;
+
+		i2cmux_0: i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+			/* Clock generator TI I2C chip, 27MHz . */
+			clockgen: cdce925pw@64 {
+				compatible = "ti,cdce925";
+				reg = <0x64>;
+				clocks = <&xtal_27Mhz>;
+				xtal-load-pf = <5>; /* Crystal load-capacitor value */
+				#clock-cells = <1>;
+				/* PLL options, optional. For example to get SSC 1%
+				 * centered, supply the following extra data:
+				 *PLL2 {
+				 *	spread-spectrum = <4>;
+				 *	spread-spectrum-center;
+				 *};
+				 */
+			};
+		};
+		i2cmux_1: i2c@1 {
+			/* HDMI O - Connects directly to HDMI connector */
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <1>;
+		};
+		i2cmux_2: i2c@2 {
+			/* GEN */
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <2>;
+			/* GPIO expander */
+			gpioexf: pca953x@20 {
+				compatible = "nxp,pca9534";
+				reg = <0x20>;
+				gpio-line-names = "USB_OTG_OC_N", "USB_OTG_EN_N", "POWER_BTN", "HDMI_IN_HPD",
+					"", "POWER_KILL_N", "MAIN_POWER", "";
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				/* IRQ line connected to MIO 15 */
+				interrupt-parent = <&gpio0>;
+				interrupts = <15 0x8>; /* GPIO 15, IRQ_TYPE_LEVEL_LOW */
+			};
+			/* EEPROM (v3r1 board)*/
+			eepromc: eepromc@50 {
+				compatible = "at24,24c04";
+				reg = <0x50>;
+			};
+			/* Programmable clock */
+			si514: si514@55 {
+				reg = <0x55>;
+				#clock-cells = <0>;
+				compatible = "silabs,si514";
+			};
+		};
+		i2cmux_3: i2c@3 {
+			/* HDMI IN */
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <3>;
+		};
+	};
+};
+
+/* USB connected through MIO and GPIO expander */
+&usb0 {
+	status = "okay";
+	vbus-supply = <&reg_usb0_vbus>;
+	dr_mode = "otg";
+	usb-phy = <&usb_phy0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usb0_default>;
+};
+
+&pinctrl0 {
+	pinctrl_usb0_default: usb0-default {
+		mux {
+			groups = "usb0_0_grp";
+			function = "usb0";
+		};
+
+		conf {
+			groups = "usb0_0_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+
+		conf-rx {
+			pins = "MIO29", "MIO31", "MIO36";
+			bias-high-impedance;
+		};
+
+		conf-tx {
+			pins = "MIO28", "MIO30", "MIO32", "MIO33", "MIO34",
+			       "MIO35", "MIO37", "MIO38", "MIO39";
+			bias-disable;
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miami-florida.dtsi b/arch/arm/boot/dts/topic-miami-florida.dtsi
new file mode 100644
index 000000000000..1b79b5fc2c9a
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida.dtsi
@@ -0,0 +1,27 @@
+/include/ "topic-miami-florida-mio.dtsi"
+
+/* Common components for ALL florida boards */
+
+/* Ethernet PHY via EMIO. Logic contains a GMII to RGMII conversion, so
+ * the emac should see a GMII interface */
+&gem0 {
+	status = "okay";
+	phy-handle = <&gmiitorgmii>;
+	phy-mode = "gmii";
+
+	phy3: phy@3 {
+		/* PHY on the board */
+		device_type = "ethernet-phy";
+		reg = <0x3>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <65 0x8>; /* GPIO 65 (EMIO 11), IRQ_TYPE_LEVEL_LOW */
+		rxc-skew-ps = <1080>; /* Skew RX clock pad by 180 ps */
+		txc-skew-ps = <840>;  /* Skew TX clock pad by -60 ps */
+	};
+
+	gmiitorgmii: gmiitorgmii@8 {
+		compatible = "xlnx,gmii-to-rgmii-1.0";
+		reg = <0x8>;
+		phy-handle = <&phy3>;
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miami.dts b/arch/arm/boot/dts/topic-miami.dts
new file mode 100644
index 000000000000..112bab6be313
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami.dts
@@ -0,0 +1,3 @@
+/dts-v1/;
+/include/ "topic-miami.dtsi"
+
diff --git a/arch/arm/boot/dts/topic-miami.dtsi b/arch/arm/boot/dts/topic-miami.dtsi
new file mode 100644
index 000000000000..d20d2d673818
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami.dtsi
@@ -0,0 +1,331 @@
+/include/ "zynq-7000.dtsi"
+
+/* Remove broken I2C driver and replace with bitbang GPIO */
+/delete-node/ &i2c0;
+/delete-node/ &i2c1;
+
+/ {
+	model = "Topic Zynq Miami";
+
+	interrupt-parent = <&intc>;
+
+	/* Arrange that UART0 is ttyPS0 */
+	aliases: aliases {
+		serial0 = &uart0;
+		serial1 = &uart1;
+		ethernet0 = &gem0;
+		mmc0 = &sdhci0;
+		mmc1 = &sdhci1;
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x000000000 0x20000000>;
+	};
+
+	chosen {
+		/* Boot using squashfs system, nice for QSPI NOR flash */
+		bootargs = "console=ttyPS0,115200 root=/dev/mtdblock5 ro rootfstype=squashfs rootwait quiet";
+		/* bootargs = "console=ttyPS0,115200 ubi.mtd=qspi-rootfs root=ubi0:qspi-rootfs rw rootfstype=ubifs rootwait"; */
+		linux,stdout-path = "/amba@0/uart@e0000000";
+	};
+
+	/* Represents the 3v3 supply on the Miami SOM. */
+	reg_3v3_miami: regulator_3v3_miami {
+		compatible = "regulator-fixed";
+		regulator-name = "3v3-miami";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
+	leds: leds {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_leds_default>;
+		led_alive: alive {
+			label = "alive:green";
+			gpios = <&gpio0 25 0>;
+			linux,default-trigger = "heartbeat";
+		};
+	};
+
+	i2c0: gpios-i2c@50 {
+		bus-id = <0>;
+		compatible = "i2c-gpio";
+		gpios = <&gpio0 51 0 /* sda */
+			 &gpio0 50 0 /* scl */
+			>;
+		i2c-gpio,delay-us = <3>;	/* 333 kHz (?) */
+		#address-cells = <1>;
+		#size-cells = <0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_i2c0_default>;
+	};
+
+	i2c1: gpios-i2c@52 {
+		bus-id = <1>;
+		compatible = "i2c-gpio";
+		gpios = <&gpio0 53 0 /* sda */
+			 &gpio0 52 0 /* scl */
+			>;
+		i2c-gpio,delay-us = <3>;	/* 333 kHz (?) */
+		#address-cells = <1>;
+		#size-cells = <0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_i2c1_default>;
+	};
+};
+
+/* Disable the ARM timer. This clocksource does not compensate for frequency
+   scaling. The TTC timer does this, but has a lower rating so it would not
+   get selected as system clocksource.
+   Alternatives to disabling this clock are setting the source in sysfs or
+   adding clocksource=ttc_clocksource to the kernel bootargs */
+&global_timer {
+	status = "disabled";
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&uart1 {
+	status = "okay";
+};
+
+&sdhci0 {
+	disable-wp; /* We don't have a write-protect detection */
+	status = "okay";
+};
+
+
+&i2c1 {
+	/* AD converter for voltage monitoring */
+	monitorvolt: ad799x@29 {
+		compatible = "ad7999";
+		reg = <0x29>;
+		vcc-supply = <&reg_3v3_miami>;
+		vref-supply = <&reg_3v3_miami>;
+	};
+	/* GPIO expander */
+	gpioex: pca953x@41 {
+		compatible = "nxp,pca9536";
+		reg = <0x41>;
+		gpio-line-names = "USB_RESET", "VTT_SHDWN_N", "V_PRESENT", "DEBUG_PRESENT";
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+	/* Current monitoring chip */
+	monitorcurrent: ltc2990@4c {
+		compatible = "lltc,ltc2990";
+		reg = <0x4c>;
+		lltc,meas-mode = <6 3>; /* Current sensor mode */
+	};
+	/* EEPROM */
+	eeprom: eeprom@50 {
+		compatible = "at24,24c04";
+		reg = <0x50>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		/* NVMEM entries */
+		gem0_mac: mac@1e8 {
+			reg = <0x1e8 6>;
+		};
+		gem1_mac: mac@1ee {
+			reg = <0x1ee 6>;
+		};
+		miamiserial: serial@1f4 {
+			reg = <0x1f4 4>;
+		};
+		dyplolicense: dyplolicense@1f8 {
+			reg = <0x1f8 8>;
+		};
+	};
+	/* IO Voltage controller */
+	ltc3562: ltc3562@65 {
+		compatible = "lltc,ltc3562";
+		reg = <0x65>; /* 0b1100101 */
+		regulators {
+			R400B_reg: R400B {
+				/* Supplies VCCO2 to bank 34 */
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-boot-on;
+				regulator-always-on;
+				lltc,operating-mode = <1>;
+			};
+			R600B_reg: R600B {
+				/* Supplies VCCO0 to Bank 13 */
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+			/* R400A Not connected */
+			R600A_reg: R600A {
+				/* Supplies VCCO1 to bank 35 */
+				/* We actually want 1v8, but the wiring of
+				 * the regulator is weird so you can only get
+				 * 1768mV */
+				regulator-min-microvolt = <1768000>;
+				regulator-max-microvolt = <1768000>;
+				lltc,fb-voltage-divider = <316 100>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+		};
+	};
+};
+
+&nand0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_smc0_nand_default>;
+
+	/* Micron MT29F2G16ABBFAH4 timings */
+	arm,nand-cycle-t0 = <0x4>;
+	arm,nand-cycle-t1 = <0x4>;
+	arm,nand-cycle-t2 = <0x1>;
+	arm,nand-cycle-t3 = <0x2>;
+	arm,nand-cycle-t4 = <0x2>;
+	arm,nand-cycle-t5 = <0x2>;
+	arm,nand-cycle-t6 = <0x4>;
+
+	#address-cells = <0x1>;
+	#size-cells = <0x1>;
+	partition@nand-all {
+		label = "nand-all";
+		reg = <0x0 0x10000000>; /* 256 MB */
+	};
+	/* to mount as UBI:
+	   ubiattach /dev/ubi_ctrl -m 0
+	   ubimkvol /dev/ubi0 -N nand -m
+	   mkdir /media/nand
+	   mount -t ubifs /dev/ubi0_0 /media/nand
+	 */
+};
+
+&usb0 {
+	dr_mode = "otg";
+};
+
+&qspi {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_qspi0_default>;
+	is-dual = <0>;
+	num-cs = <1>;
+	primary_flash: ps7-qspi@0 {
+		compatible = "jedec,spi-nor";
+		/* Max frequency for quad read is 108MHz according to data sheet */
+		spi-max-frequency = <108000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <4>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0x0>;
+		partition@qspi-u-boot-spl {
+			label = "qspi-u-boot-spl";
+			reg = <0x00000 0x20000>;
+		};
+		partition@qspi-u-boot-img {
+			label = "qspi-u-boot-img";
+			reg = <0x20000 0xC0000>;
+		};
+		partition@qspi-u-boot-env {
+			label = "qspi-u-boot-env";
+			reg = <0xE0000 0x20000>;
+		};
+		partition@qspi-rootfs {
+			label = "qspi-rootfs";
+			reg = <0x100000 0x1F00000>;
+		};
+		/* Everything */
+		partition@qspi-all {
+			label = "qspi-all";
+			reg = <0x0 0x2000000>;
+		};
+	};
+};
+
+&watchdog0 {
+	reset-on-timeout;
+};
+
+&gem0 {
+	/* MAC address stored in NVMEM */
+	nvmem-cells = <&gem0_mac>;
+	nvmem-cell-names = "mac-address";
+};
+
+&gem1 {
+	/* MAC address stored in NVMEM */
+	nvmem-cells = <&gem1_mac>;
+	nvmem-cell-names = "mac-address";
+};
+
+&pinctrl0 {
+	pinctrl_qspi0_default: qspi0_default {
+		mux {
+			groups = "qspi0_0_grp";
+			function = "qspi0";
+		};
+		mux-fbclk {
+			groups = "qspi_fbclk_grp";
+			function = "qspi_fbclk";
+		};
+		conf {
+			groups = "qspi0_0_grp", "qspi_fbclk_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+			bias-disable;
+		};
+	};
+
+	pinctrl_smc0_nand_default: smc0_nand_default {
+		mux {
+			groups = "smc0_nand_grp";
+			function = "smc0_nand";
+		};
+	};
+
+	pinctrl_leds_default: leds-default {
+		mux {
+			groups = "gpio0_25_grp";
+			function = "gpio0";
+		};
+
+		conf {
+			groups = "gpio0_25_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+	};
+
+	pinctrl_i2c0_default: i2c0-default {
+		mux {
+			groups = "gpio0_50_grp", "gpio0_51_grp";
+			function = "gpio0";
+		};
+
+		conf {
+			groups = "gpio0_50_grp", "gpio0_51_grp";
+			bias-pull-up;
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+	};
+
+	pinctrl_i2c1_default: i2c1-default {
+		mux {
+			groups = "gpio0_52_grp", "gpio0_53_grp";
+			function = "gpio0";
+		};
+
+		conf {
+			groups = "gpio0_52_grp", "gpio0_53_grp";
+			bias-pull-up;
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miamilite.dts b/arch/arm/boot/dts/topic-miamilite.dts
new file mode 100644
index 000000000000..6a827e6d3e33
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miamilite.dts
@@ -0,0 +1,3 @@
+/dts-v1/;
+/include/ "topic-miami.dtsi"
+/include/ "topic-miamilite.dtsi"
diff --git a/arch/arm/boot/dts/topic-miamilite.dtsi b/arch/arm/boot/dts/topic-miamilite.dtsi
new file mode 100644
index 000000000000..8b352dda67bc
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miamilite.dtsi
@@ -0,0 +1,51 @@
+/* The "lite" version has less I/O pins and less periferals */
+
+/* Remove missing periferals */
+/delete-node/ &monitorcurrent;
+/delete-node/ &monitorvolt;
+/delete-node/ &ltc3562;
+/delete-node/ &gpioex;
+
+/* Alive LED is on MIO9 */
+&led_alive {
+	gpios = <&gpio0 9 1>; /* Active low */
+};
+
+&pinctrl_leds_default {
+	mux {
+		groups = "gpio0_9_grp";
+		function = "gpio0";
+	};
+	conf {
+		groups = "gpio0_9_grp";
+		slew-rate = <0>;
+		io-standard = <1>;
+	};
+};
+
+/* I2C bus 1 using pins 48+49 */
+&i2c1 {
+	gpios = <&gpio0 49 0 /* sda */
+		 &gpio0 48 0 /* scl */
+		>;
+   
+	/* 32Kb EEPROM */
+	eeprom: eeprom@50 {
+		compatible = "at24,24c32";
+		reg = <0x50>;
+	};
+};
+
+&pinctrl_i2c1_default {
+	mux {
+		groups = "gpio0_48_grp", "gpio0_49_grp";
+		function = "gpio0";
+	};
+
+	conf {
+		groups = "gpio0_48_grp", "gpio0_49_grp";
+		bias-pull-up;
+		slew-rate = <0>;
+		io-standard = <1>;
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miamiplus.dts b/arch/arm/boot/dts/topic-miamiplus.dts
new file mode 100644
index 000000000000..cc6eb2990e64
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miamiplus.dts
@@ -0,0 +1,4 @@
+/dts-v1/;
+/include/ "topic-miami.dtsi"
+/include/ "topic-miamiplus.dtsi"
+
diff --git a/arch/arm/boot/dts/topic-miamiplus.dtsi b/arch/arm/boot/dts/topic-miamiplus.dtsi
new file mode 100644
index 000000000000..c63b30266fa3
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miamiplus.dtsi
@@ -0,0 +1,97 @@
+/* the "plus" version has much more I/O pins, but lacks some periferals,
+ * and the NAND was replaced by a dual-QSPI NOR setup. */
+
+/* The miamiplus contains a speedgrade-2 device and runs at 800MHz */
+&cpu0 {
+	operating-points = <
+		/* kHz    uV */
+		800000  1000000
+		400000  1000000
+	>;
+};
+
+/* Remove missing periferals */
+/delete-node/ &monitorcurrent;
+/delete-node/ &ltc3562;
+
+/* Different voltage monitoring IC */
+&monitorvolt {
+	compatible = "ad7997";
+	reg = <0x21>;
+	/* TODO: gpios */
+};
+
+/* Dual QSPI configuration */
+&qspi {
+	is-dual = <1>;
+};
+
+/* Dual flash, also double the capacity so adjust rootfs and "all" partitions */
+&primary_flash {
+	partition@qspi-rootfs {
+		label = "qspi-rootfs";
+		reg = <0x500000 0x3b00000>;
+	};
+	partition@qspi-all {
+		label = "qspi-all";
+		reg = <0x0 0x4000000>;
+	};
+};
+
+/* The plus SOMs have a programmable clock for the high-speed tranceivers */
+&i2c1 {
+	si514s: si514@55 {
+		reg = <0x55>;
+		#clock-cells = <0>;
+		compatible = "silabs,si514";
+		clock-output-names = "si514miami";
+	};
+	max31760: max31760@57 {
+		reg = <0x57>; /* Address pins "111" hence address 1010111b */
+		compatible = "maxim,max31760";
+	};
+};
+
+/* Extra user board LED on GPIO expander */
+&leds {
+	led_user2: user2 {
+		label = "user2:green";
+		gpios = <&gpioex 3 0>;
+	};
+};
+/* Alive LED is on MIO7 instead of 25 */
+&led_alive {
+	gpios = <&gpio0 7 0>;
+};
+&pinctrl_leds_default {
+	mux {
+		groups = "gpio0_7_grp";
+		function = "gpio0";
+	};
+	conf {
+		groups = "gpio0_7_grp";
+		slew-rate = <0>;
+		io-standard = <1>;
+	};
+};
+
+/* I2C bus 1 using pins 48+49 */
+&i2c1 {
+	gpios = <&gpio0 49 0 /* sda */
+		 &gpio0 48 0 /* scl */
+		>;
+};
+
+&pinctrl_i2c1_default {
+	mux {
+		groups = "gpio0_48_grp", "gpio0_49_grp";
+		function = "gpio0";
+	};
+
+	conf {
+		groups = "gpio0_48_grp", "gpio0_49_grp";
+		bias-pull-up;
+		slew-rate = <0>;
+		io-standard = <1>;
+	};
+};
diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
new file mode 100644
index 000000000000..2b3ed5644ec5
--- /dev/null
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
@@ -0,0 +1,282 @@
+/*
+ * dts file for Topic Miami MPSoC
+ *
+ * (C) Copyright 2017, Topic Embedded Products BV
+ * Mike Looijmans <mike.looijmans@topic.nl>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ */
+
+#include "zynqmp-topic-miamimp.dts"
+
+/ {
+	/* Regulator for the 5V USB VBUS power line. This is connected
+	 * to the second output of the GPIO expander. */
+	reg_usb0_vbus: regulator_usb {
+		compatible = "regulator-fixed";
+		regulator-name = "usb0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&gpioexf 1 0>;
+		enable-active-high;
+	};
+
+	xtal_27Mhz: clock_xtal_27Mhz {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <27000000>;
+	};
+
+	gpio_keys: gpio-keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		power {
+			label = "power";
+			gpios = <&gpioexf 2 1>; /* Input 2, active low */
+			linux,code = <116>; /* KEY_POWER */
+			gpio-key,wakeup;
+		};
+	};
+
+	gpio-poweroff {
+		compatible = "gpio-poweroff";
+		gpios = <&gpioexf 5 1>; /* Output 5, active low */
+		input;
+		force-pm-power-off; /* Replace the PSCI power-off callback */
+	};
+
+	charger: battery-charger {
+		compatible = "gpio-charger";
+		charger-type = "mains";
+		gpios = <&gpioexf 6 1>; /* Pin 6 of IO expander, active low */
+	};
+};
+
+&leds {
+	traffic_led_d4 { /* Red */
+		label = "d4:red";
+		gpios = <&gpio 82 0>; /* EMIO 4 */
+		linux,default-trigger = "mmc1";
+	};
+	traffic_led_d3 { /* Orange */
+		label = "d3:orange";
+		gpios = <&gpio 81 0>; /* EMIO 3 */
+		linux,default-trigger = "mmc0";
+	};
+	traffic_led_d2 { /* Green */
+		label = "d2:green";
+		gpios = <&gpio 80 0>; /* EMIO 2 */
+	};
+	traffic_led_d6 { /* Green */
+		label = "d6:green";
+		gpios = <&gpio 79 0>; /* EMIO 1 */
+	};
+	traffic_led_d5 { /* Green */
+		label = "d5:green";
+		gpios = <&gpio 78 0>; /* EMIO 0 */
+	};
+};
+
+/* SD card with level shifter */
+&sdhci1 {
+	status = "okay";
+	xlnx,mio_bank = <1>;
+	no-1-8-v; /* No high-speed capability */
+	disable-wp; /* We don't have a write-protect detection */
+};
+
+&uart1 {
+	status = "okay";
+};
+
+/* Setup for SATA serdes clock */
+&si514s {
+	assigned-clock-rates = <150000000>;
+};
+
+/* I2C bus layout */
+&i2c0 {
+	i2cmux: i2cswitch@70 {
+		compatible = "nxp,pca9546";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0x70>;
+
+		i2cmux_0: i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+			/* Clock generator TI I2C chip, 27MHz . */
+			clockgen: cdce925pw@64 {
+				compatible = "ti,cdce925";
+				reg = <0x64>;
+				clocks = <&xtal_27Mhz>;
+				xtal-load-pf = <5>; /* Crystal load-capacitor value */
+				#clock-cells = <1>;
+				/* PLL options, optional. For example to get SSC 1%
+				 * centered, supply the following extra data:
+				 *PLL2 {
+				 *	spread-spectrum = <4>;
+				 *	spread-spectrum-center;
+				 *};
+				 */
+			};
+		};
+		i2cmux_1: i2c@1 {
+			/* HDMI O - Connects directly to HDMI connector */
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <1>;
+		};
+		i2cmux_2: i2c@2 {
+			/* GEN */
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <2>;
+			/* GPIO expander */
+			gpioexf: pca953x@20 {
+				compatible = "nxp,pca9534";
+				reg = <0x20>;
+				vcc-supply = <&reg_3v3_miami>;
+				gpio-line-names = "USB_OTG_OC_N", "USB_OTG_EN_N", "POWER_BTN", "HDMI_IN_HPD",
+					"", "POWER_KILL_N", "MAIN_POWER", "";
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				/* IRQ line connected to MIO ??? */
+				interrupt-parent = <&gpio>;
+				interrupts = <36 0x8>; /* GPIO 36, IRQ_TYPE_LEVEL_LOW */
+			};
+			/* Touch */
+			touch: ad7879@2f {
+				status = "disabled"; /* No display panel support, so touch is pointless */
+				compatible = "ad7879";
+				reg = <0x2F>;
+				interrupt-parent = <&gpio>;
+				interrupts = <94 0x2>; /* GPIO 78+16 (EMIO 16), falling edge */
+				touchscreen-swapped-x-y;
+				adi,resistance-plate-x = <620>;
+				touchscreen-max-pressure = <4096>;
+				adi,first-conversion-delay = /bits/ 8 <3>;
+				adi,acquisition-time = /bits/ 8 <1>;
+				adi,median-filter-size = /bits/ 8 <2>;
+				adi,averaging = /bits/ 8 <1>;
+				adi,conversion-interval = /bits/ 8 <255>;
+			};
+			/* EEPROM */
+			eepromc: eepromc@50 {
+				compatible = "at24,24c04";
+				reg = <0x50>;
+			};
+			/* Programmable clock */
+			si514c: si514@55 {
+				reg = <0x55>;
+				#clock-cells = <0>;
+				compatible = "silabs,si514";
+			};
+			/* Battery gauge */
+			fuelgauge: battery-gauge@64 {
+				compatible = "ltc2943";
+				reg = <0x64>;
+				lltc,resistor-sense = <15>; /* 15 mOhm */
+				lltc,prescaler-exponent = <5>; /* 2^(2*5) = 1024 */
+				power-supplies = <&charger>;
+			};
+		};
+		i2cmux_3: i2c@3 {
+			/* HDMI IN */
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <3>;
+		};
+	};
+};
+
+/* USB connected through MIO and GPIO expander */
+&usb0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usb0_default>;
+	/* see: https://forums.xilinx.com/t5/Embedded-Linux/Zynqmp-USB2-0/td-p/790522/page/2 */
+	/delete-property/ clocks;
+	/delete-property/ clock-names;
+	clocks = <&zynqmp_clk USB0_BUS_REF>;
+	clock-names = "bus_clk";
+};
+
+&dwc3_0 {
+	status = "okay";
+	dr_mode = "otg"; /* "host", "peripheral" or "otg" */
+	vbus-supply = <&reg_usb0_vbus>;
+	maximum-speed = "high-speed"; /* super-speed not available on carrier */
+	/* see: https://forums.xilinx.com/t5/Embedded-Linux/Zynqmp-USB2-0/td-p/790522/page/2 */
+	snps,dis_u2_susphy_quirk;
+	snps,dis_u3_susphy_quirk;
+	/delete-property/ phy-names;
+	/delete-property/ phys;
+	usb-phy = <&usb_phy0>;
+	ulpi {
+		phy {
+			compatible = "smsc,usb3340";
+		};
+	};
+};
+
+&pinctrl0 {
+	status = "okay";
+
+	pinctrl_sdhci1_default: sdhci1-default {
+		mux {
+			groups = "sdio1_4bit_0_1_grp";
+			function = "sdio1";
+		};
+
+		conf {
+			groups = "sdio1_4bit_0_1_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+			bias-disable;
+		};
+
+		mux-cd {
+			groups = "sdio1_0_cd_grp";
+			function = "sdio1_cd";
+		};
+
+		conf-cd {
+			groups = "sdio1_0_cd_grp";
+			bias-high-impedance;
+			bias-pull-up;
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_usb0_default: usb0-default {
+		mux {
+			groups = "usb0_0_grp";
+			function = "usb0";
+		};
+
+		conf {
+			groups = "usb0_0_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+
+		conf-rx {
+			pins = "MIO52", "MIO53", "MIO55";
+			bias-high-impedance;
+		};
+
+		conf-tx {
+			pins = "MIO54", "MIO56", "MIO57", "MIO58", "MIO59",
+			       "MIO60", "MIO61", "MIO62", "MIO63";
+			bias-disable;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-test.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-test.dts
new file mode 100644
index 000000000000..7739d2e82a67
--- /dev/null
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-test.dts
@@ -0,0 +1,187 @@
+/*
+ * dts file for Topic Miami MPSoC
+ *
+ * (C) Copyright 2017, Topic Embedded Products BV
+ * Mike Looijmans <mike.looijmans@topic.nl>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ */
+
+#include "zynqmp-topic-miamimp.dts"
+
+/* Common components for MIO parts of Florida boards. These are
+ * accessible even without programming any logic. */
+/ {
+	/* Regulator for the 5V USB VBUS power line. This is connected
+	 * to the second output of the GPIO expander. */
+	reg_usb0_vbus: regulator_usb {
+		compatible = "regulator-fixed";
+		regulator-name = "usb0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&gpioexf 1 0>;
+		enable-active-high;
+		regulator-always-on; /* USB host mode */
+	};
+
+	gpio_keys: gpio-keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		power {
+			label = "power";
+			gpios = <&gpioexf 2 1>; /* Input 2, active low */
+			linux,code = <116>; /* KEY_POWER */
+			gpio-key,wakeup;
+		};
+		/* On-board push buttons in reference design */
+		s2 {
+			label = "s2";
+			gpios = <&gpio 95 1>; /* EMIO 17, active low */
+			linux,code = <60>; /* KEY_F2 */
+			gpio-key,wakeup;
+		};
+		s3 {
+			label = "s3";
+			gpios = <&gpio 96 1>; /* EMIO 18, active low */
+			linux,code = <61>; /* KEY_F3 */
+			gpio-key,wakeup;
+		};
+	};
+
+	gpio-poweroff {
+		compatible = "gpio-poweroff";
+		gpios = <&gpioexf 5 1>; /* Output 5, active low */
+		input;
+	};
+};
+
+/* SD card with level shifter */
+&sdhci1 {
+	status = "okay";
+	no-1-8-v;	/* for 1.0 silicon */
+	xlnx,mio_bank = <1>;
+	disable-wp; /* We don't have a write-protect detection */
+};
+
+&serdes {
+	status = "okay";
+};
+
+&uart1 {
+	status = "okay";
+};
+
+/* I2C bus layout */
+&i2c0 {
+	/* GPIO expander 3v3 */
+	gpioexf: pca953x@20 {
+		compatible = "nxp,pca9534";
+		reg = <0x20>;
+		/* 0=USB_OTG_OC_N 1=USB_OTG_EN_N 2=n.c. ...*/
+		gpio-controller;
+		#gpio-cells = <2>;
+		interrupt-controller;
+		interrupt-parent = <&gpio>;
+	};
+
+	/* Programmable clock, to clock-capable pins on FPGA */
+	si514c: si514@carrier {
+		reg = <0x55>;
+		#clock-cells = <0>;
+		compatible = "silabs,si514";
+		clock-output-names = "si514carrier";
+	};
+
+	/* GPIO expanders 1v8 */
+	gpiotst1: pca953x@21 {
+		compatible = "nxp,pca9534";
+		reg = <0x21>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+	gpiotst2: pca953x@22 {
+		compatible = "nxp,pca9534";
+		reg = <0x22>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+	gpiotst3: pca953x@23 {
+		compatible = "nxp,pca9534";
+		reg = <0x23>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+};
+
+/* USB connected through MIO and GPIO expander */
+&usb0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usb0_default>;
+};
+&dwc3_0 {
+	status = "okay";
+	dr_mode = "host"; /* "host", "peripheral" or "otg" */
+	maximum-speed = "high-speed"; /* super-speed not available on carrier */
+	vbus-supply = <&reg_usb0_vbus>; /* This apparently no longer works */
+	usb-phy = <&usb_phy0>;
+};
+
+&pinctrl0 {
+	status = "okay";
+
+	pinctrl_sdhci1_default: sdhci1-default {
+		mux {
+			groups = "sdio1_4bit_0_1_grp";
+			function = "sdio1";
+		};
+
+		conf {
+			groups = "sdio1_4bit_0_1_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+			bias-disable;
+		};
+
+		mux-cd {
+			groups = "sdio1_0_cd_grp";
+			function = "sdio1_cd";
+		};
+
+		conf-cd {
+			groups = "sdio1_0_cd_grp";
+			bias-high-impedance;
+			bias-pull-up;
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_usb0_default: usb0-default {
+		mux {
+			groups = "usb0_0_grp";
+			function = "usb0";
+		};
+
+		conf {
+			groups = "usb0_0_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+
+		conf-rx {
+			pins = "MIO52", "MIO53", "MIO55";
+			bias-high-impedance;
+		};
+
+		conf-tx {
+			pins = "MIO54", "MIO56", "MIO57", "MIO58", "MIO59",
+			       "MIO60", "MIO61", "MIO62", "MIO63";
+			bias-disable;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts
new file mode 100644
index 000000000000..87fc10e99036
--- /dev/null
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts
@@ -0,0 +1,377 @@
+/*
+ * dts file for Topic Miami MPSoC
+ *
+ * (C) Copyright 2017, Topic Embedded Products BV
+ * Mike Looijmans <mike.looijmans@topic.nl>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ */
+
+/dts-v1/;
+
+#include "zynqmp.dtsi"
+#include "zynqmp-clk-ccf.dtsi"
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/pinctrl/pinctrl-zynqmp.h>
+#include <dt-bindings/phy/phy.h>
+
+/ {
+	model = "Topic Miami MPSoC";
+	compatible = "topic,miamimp", "xlnx,zynqmp";
+
+	aliases {
+		ethernet0 = &gem3;
+		gpio0 = &gpio;
+		i2c0 = &i2c0;
+		i2c1 = &i2c1;
+		mmc0 = &sdhci0;
+		mmc1 = &sdhci1;
+		rtc0 = &rtc;
+		serial0 = &uart0;
+		serial1 = &uart1;
+		spi0 = &qspi;
+		usb0 = &usb0;
+	};
+
+	chosen {
+		bootargs = "earlycon root=/dev/mmcblk1p2 rw rootfstype=ext4 rootwait";
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+
+	/* Represents the 3v3 supply on the Miami SOM. */
+	reg_3v3_miami: regulator_3v3_miami {
+		compatible = "regulator-fixed";
+		regulator-name = "3v3-miami";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
+	leds: leds {
+		compatible = "gpio-leds";
+		heartbeat_led {
+			label = "heartbeat:green";
+			gpios = <&gpio 35 0>;
+			linux,default-trigger = "heartbeat";
+		};
+		user_led_2 {
+			label = "user2:green";
+			gpios = <&gpioex 3 0>;
+		};
+	};
+
+	/* USB PHY reset IO pin. Carrier board's USB PHY can use this. */
+	usb_phy0: phy0 {
+		compatible = "usb-nop-xceiv";
+		#phy-cells = <0>;
+		vcc-supply = <&reg_3v3_miami>;
+		reset-gpios = <&gpioex 0 GPIO_ACTIVE_LOW>;
+	};
+
+	/* Make temperature readouts show up in hwmon applications */
+	iio-hwmon-ams-temp {
+		compatible = "iio-hwmon";
+		io-channels = <&xilinx_ams 7>, <&xilinx_ams 8>, <&xilinx_ams 20>;
+		io-channel-names = "ps_temp", "remote_temp", "pl_temp";
+	};
+};
+
+&gpio {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio_default>;
+};
+
+&gpu {
+	status = "okay";
+};
+
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <400000>;
+	pinctrl-names = "default", "gpio";
+	pinctrl-0 = <&pinctrl_i2c0_default>;
+	pinctrl-1 = <&pinctrl_i2c0_gpio>;
+	scl-gpios = <&gpio 26 GPIO_ACTIVE_HIGH>;
+	sda-gpios = <&gpio 27 GPIO_ACTIVE_HIGH>;
+};
+
+&i2c1 {
+	status = "okay";
+	clock-frequency = <400000>;
+	pinctrl-names = "default", "gpio";
+	pinctrl-0 = <&pinctrl_i2c1_default>;
+	pinctrl-1 = <&pinctrl_i2c1_gpio>;
+	scl-gpios = <&gpio 32 GPIO_ACTIVE_HIGH>;
+	sda-gpios = <&gpio 33 GPIO_ACTIVE_HIGH>;
+
+	/* GPIO expander */
+	gpioex: pca953x@41 {
+		compatible = "nxp,pca9536";
+		reg = <0x41>;
+		vcc-supply = <&reg_3v3_miami>;
+		gpio-line-names = "USB_RESET", "VTT_SHDWN_N", "V_PRESENT", "LED2";
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+	/* EEPROM */
+	eeprom: eeprom@50 {
+		compatible = "at24,24c04";
+		reg = <0x50>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		gem0_mac: mac@1e8 {
+			reg = <0x1e8 6>;
+		};
+		gem1_mac: mac@1ee {
+			reg = <0x1ee 6>;
+		};
+		miamiserial: serial@1f4 {
+			reg = <0x1f4 4>;
+		};
+		dyplolicense: dyplolicense@1f8 {
+			reg = <0x1f8 8>;
+		};
+	};
+	/* Programmable clock (for logic) */
+	si514s: si514@55 {
+		reg = <0x55>;
+		#clock-cells = <0>;
+		compatible = "silabs,si514";
+		clock-output-names = "si514miami";
+	};
+	/* AD converter for voltage monitoring */
+	monitorvolt: ad799x@21 {
+		compatible = "ad7997";
+		reg = <0x21>;
+		vcc-supply = <&reg_3v3_miami>;
+		vref-supply = <&reg_3v3_miami>;
+	};
+};
+
+&qspi {
+	status = "okay";
+	is-dual = <1>;
+	multi-die = <1>;
+	flash@0 {
+		compatible = "st,m25p80", "n25q256a";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0x0>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <4>;
+		spi-max-frequency = <166000000>;
+		is-dual = <1>;
+		multi-die = <1>;
+		partition@qspi-boot-bin {
+			label = "qspi-boot-bin";
+			reg = <0x00000 0x60000>;
+		};
+		partition@qspi-u-boot-itb {
+			label = "qspi-u-boot-itb";
+			reg = <0x60000 0x100000>;
+		};
+		partition@qspi-u-boot-env {
+			label = "qspi-u-boot-env";
+			reg = <0x160000 0x20000>;
+		};
+		partition@qspi-rootfs {
+			label = "qspi-rootfs";
+			reg = <0x180000 0x7e80000>;
+		};
+		/* Everything */
+		partition@qspi-all {
+			label = "qspi-all";
+			reg = <0x0 0x8000000>;
+		};
+	};
+};
+
+&rtc {
+	status = "disabled"; /* Doesn't work unless you connect VBAT */
+};
+
+/* eMMC device */
+&sdhci0 {
+	status = "okay";
+	non-removable;
+	disable-wp; /* We don't have a write-protect detection */
+	bus-width = <8>;
+	xlnx,mio_bank = <0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sdhci0_default>;
+};
+
+&serdes {
+	status = "okay";
+};
+
+&uart0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart0_default>;
+};
+
+&watchdog0 {
+	status = "okay";
+	timeout-sec = <5>;
+	reset-on-timeout;
+};
+
+&xilinx_ams {
+	status = "okay";
+};
+
+&ams_ps {
+	status = "okay";
+};
+
+&ams_pl {
+	status = "okay";
+};
+
+&pinctrl0 {
+	status = "okay";
+
+	pinctrl_i2c0_default: i2c0-default {
+		mux {
+			groups = "i2c0_6_grp";
+			function = "i2c0";
+		};
+
+		conf {
+			groups = "i2c0_6_grp";
+			bias-pull-up;
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_i2c0_gpio: i2c0-gpio {
+		mux {
+			groups = "gpio0_26_grp", "gpio0_27_grp";
+			function = "gpio0";
+		};
+
+		conf {
+			groups = "gpio0_26_grp", "gpio0_27_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_i2c1_default: i2c1-default {
+		mux {
+			groups = "i2c1_8_grp";
+			function = "i2c1";
+		};
+
+		conf {
+			groups = "i2c1_8_grp";
+			bias-pull-up;
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_i2c1_gpio: i2c1-gpio {
+		mux {
+			groups = "gpio0_32_grp", "gpio0_33_grp";
+			function = "gpio0";
+		};
+
+		conf {
+			groups = "gpio0_32_grp", "gpio0_33_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_uart0_default: uart0-default {
+		mux {
+			groups = "uart0_7_grp";
+			function = "uart0";
+		};
+
+		conf {
+			groups = "uart0_7_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+
+		conf-rx {
+			pins = "MIO30";
+			bias-high-impedance;
+		};
+
+		conf-tx {
+			pins = "MIO31";
+			bias-disable;
+		};
+	};
+
+	pinctrl_sdhci0_default: sdhci0-default {
+		mux {
+			groups = "sdio0_0_grp";
+			function = "sdio0";
+		};
+
+		conf {
+			groups = "sdio0_0_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+			bias-disable;
+		};
+
+		mux-cd {
+			groups = "sdio0_cd_0_grp";
+			function = "sdio0_cd";
+		};
+
+		conf-cd {
+			groups = "sdio0_cd_0_grp";
+			bias-high-impedance;
+			bias-pull-up;
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_gpio_default: gpio-default {
+		mux-sw {
+			function = "gpio0";
+			groups = "gpio0_35_grp";
+		};
+
+		conf-sw {
+			groups = "gpio0_35_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+
+		mux-msp {
+			function = "gpio0";
+			groups = "gpio0_35_grp";
+		};
+
+		conf-msp {
+			groups = "gpio0_35_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+
+		conf-pull-none {
+			pins = "MIO35";
+			bias-disable;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
new file mode 100644
index 000000000000..418797ace8a8
--- /dev/null
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
@@ -0,0 +1,373 @@
+/*
+ * dts file for Topic Miami MPSoC
+ *
+ * (C) Copyright 2017, Topic Embedded Products BV
+ * Mike Looijmans <mike.looijmans@topic.nl>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ */
+
+/dts-v1/;
+
+#include "zynqmp.dtsi"
+#include "zynqmp-clk-ccf.dtsi"
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/pinctrl/pinctrl-zynqmp.h>
+#include <dt-bindings/phy/phy.h>
+
+/ {
+	model = "Topic Miami MPSoC";
+	compatible = "topic,miamiplusmp", "xlnx,zynqmp";
+
+	aliases {
+		ethernet0 = &gem2;
+		gpio0 = &gpio;
+		i2c0 = &i2c0;
+		i2c1 = &i2c1;
+		mmc0 = &sdhci0;
+		mmc1 = &sdhci1;
+		rtc0 = &rtc;
+		serial0 = &uart0;
+		serial1 = &uart1;
+		spi0 = &qspi;
+		usb0 = &usb0;
+		usb1 = &usb1;
+	};
+
+	chosen {
+		bootargs = "earlycon root=/dev/mmcblk1p2 rw rootfstype=ext4 rootwait";
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+
+	/* Represents the 3v3 supply on the SOM. */
+	reg_3v3_miami: regulator_3v3_miami {
+		compatible = "regulator-fixed";
+		regulator-name = "3v3-miami";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
+	/* Represents the 1v8 supply on the SOM. */
+	reg_1v8_miami: regulator_1v8_miami {
+		compatible = "regulator-fixed";
+		regulator-name = "1v8-miami";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+	};
+
+	/* Heartbeat led. Use separate instance to make it probe quickly */
+	leds-mio {
+		compatible = "gpio-leds";
+		heartbeat_led {
+			label = "heartbeat:green";
+			gpios = <&gpio 25 1>;
+			linux,default-trigger = "heartbeat";
+		};
+	};
+
+	/* LEDs connected to GPIO expanders and such */
+	leds: leds {
+		compatible = "gpio-leds";
+		user_led_2 {
+			label = "user2:green";
+			gpios = <&gpioex 0 1>;
+		};
+	};
+
+	/* XTAL for Si5345 */
+	xtal50MHz: xtal_50_clock {
+		compatible = "fixed-clock";
+		#clock-cells = <0x0>;
+		clock-frequency = <50000000>;
+	};
+
+	/* Make temperature readouts show up in hwmon applications */
+	iio-hwmon-ams-temp {
+		compatible = "iio-hwmon";
+		io-channels = <&xilinx_ams 7>, <&xilinx_ams 8>, <&xilinx_ams 20>;
+		io-channel-names = "ps_temp", "remote_temp", "pl_temp";
+	};
+};
+
+&gpio {
+	status = "okay";
+};
+
+&gpu {
+	status = "okay";
+};
+
+&i2c1 {
+	status = "okay";
+	clock-frequency = <400000>;
+
+	/* GPIO expander */
+	gpioex: pca9554@38 {
+		compatible = "nxp,pca9554";
+		vcc-supply = <&reg_1v8_miami>;
+		reg = <0x38>; /* PCA9554A adress starts at 0b111000 */
+		gpio-line-names = "USERLED", "", "ETH_RESET_N", "USB0_RESET_N", "USB1_RESET_N", "", "", "";
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+
+	/* EEPROM */
+	eeprom: eeprom@51 {
+		compatible = "atmel,24c32";
+		reg = <0x51>;
+		pagesize = <32>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		gem2_mac: mac@0 {
+			reg = <0x0 6>;
+		};
+	};
+
+	/* Programmable clock (for logic) */
+	si5345: si5345@68 {
+		reg = <0x68>;
+		compatible = "silabs,si5345";
+		#clock-cells = <2>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		clocks = <&xtal50MHz>;
+		clock-names = "xtal";
+		/* Interrupt connected to MIO26 (not actually used) */
+		interrupt-parent = <&gpio>;
+		interrupts = <26 IRQ_TYPE_LEVEL_LOW>;
+
+		silabs,pll-m-num = <280>; /* PLL at 280x50M = 14.0 GHz */
+		silabs,pll-m-den = <1>;
+
+		assigned-clocks = <&si5345 1 0>,
+				  <&si5345 1 1>,
+				  <&si5345 1 2>,
+				  <&si5345 1 3>,
+				  <&si5345 1 4>,
+				  <&si5345 0 0>,
+				  <&si5345 0 1>,
+				  <&si5345 0 2>,
+				  <&si5345 0 3>,
+				  <&si5345 0 4>,
+				  <&si5345 0 5>,
+				  <&si5345 0 6>,
+				  <&si5345 0 7>,
+				  <&si5345 0 8>,
+				  <&si5345 0 9>;
+		assigned-clock-parents = <0>, <0>, <0>, <0>, <0>,
+					<&si5345 1 0>, /* out 0 */
+					<&si5345 1 0>,
+					<&si5345 1 2>,
+					<&si5345 1 0>,
+					<&si5345 1 0>,
+					<&si5345 1 2>, /* out 5 */
+					<&si5345 1 0>,
+					<&si5345 1 0>,
+					<&si5345 1 1>,
+					<&si5345 1 0>;
+		assigned-clock-rates =	<400000000>, /* synth 0 */
+					<104000000>,
+					<250000000>,
+					<0>,
+					<0>,
+					<100000000>, /* out 0 */
+					<100000000>,
+					<125000000>,
+					<100000000>,
+					<100000000>,
+					< 25000000>, /* out 5 (ethernet) */
+					<100000000>,
+					<100000000>,
+					< 26000000>, /* out 8 (USB) */
+					<100000000>;
+
+		out@0 {
+			reg = <0>;
+			silabs,format = <1>; /* LVDS 3v3 */
+			silabs,common-mode = <3>;
+			silabs,amplitude = <3>;
+		};
+		out@1 {
+			reg = <1>;
+			silabs,format = <1>; /* LVDS 3v3 */
+			silabs,common-mode = <3>;
+			silabs,amplitude = <3>;
+		};
+		out@2 {
+			/* PS_MGT_D_REFCLK_1 */
+			reg = <2>;
+			silabs,format = <1>; /* LVDS 3v3 */
+			silabs,common-mode = <3>;
+			silabs,amplitude = <3>;
+			always-on;
+		};
+		out@3 {
+			/* PS_MGT_D_REFCLK_0 */
+			reg = <3>;
+			silabs,format = <1>; /* LVDS 3v3 */
+			silabs,common-mode = <3>;
+			silabs,amplitude = <3>;
+			always-on;
+		};
+		out@4 {
+			reg = <4>;
+			silabs,format = <1>; /* LVDS 3v3 */
+			silabs,common-mode = <3>;
+			silabs,amplitude = <3>;
+		};
+		out@5 {
+			/* Ethernet PHY clock */
+			reg = <5>;
+			silabs,format = <4>; /* CMOS */
+			silabs,common-mode = <3>; /* TODO: Correct values... */
+			silabs,amplitude = <3>; /* TODO: Correct values... */
+			always-on;
+		};
+		out@6 {
+			reg = <6>;
+			silabs,format = <1>; /* LVDS 3v3 */
+			silabs,common-mode = <3>;
+			silabs,amplitude = <3>;
+		};
+
+		out@7 {
+			reg = <7>;
+			silabs,format = <1>; /* LVDS 1v8 */
+			silabs,common-mode = <13>;
+			silabs,amplitude = <3>;
+		};
+
+		out@8 {
+			/* PS_MGT_D_REFCLK_3 */
+			reg = <8>;
+			silabs,format = <1>; /* LVDS 1v8 */
+			silabs,common-mode = <13>;
+			silabs,amplitude = <3>;
+			always-on;
+		};
+
+		out@9 {
+			reg = <9>;
+			silabs,format = <1>; /* LVDS 1v8 */
+			silabs,common-mode = <13>;
+			silabs,amplitude = <3>;
+		};
+	};
+};
+
+&qspi {
+	status = "okay";
+	is-dual = <1>;
+	flash@0 {
+		compatible = "st,m25p80", "n25q256a";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0x0>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <4>;
+		spi-max-frequency = <166000000>;
+		is-dual = <1>;
+		partition@qspi-boot-bin {
+			label = "qspi-boot-bin";
+			reg = <0x00000 0x60000>;
+		};
+		partition@qspi-u-boot-itb {
+			label = "qspi-u-boot-itb";
+			reg = <0x60000 0x100000>;
+		};
+		partition@qspi-u-boot-env {
+			label = "qspi-u-boot-env";
+			reg = <0x160000 0x20000>;
+		};
+		partition@qspi-rootfs {
+			label = "qspi-rootfs";
+			reg = <0x180000 0x7e80000>;
+		};
+		/* Everything */
+		partition@qspi-all {
+			label = "qspi-all";
+			reg = <0x0 0x8000000>;
+		};
+	};
+};
+
+/* eMMC device */
+&sdhci0 {
+	status = "okay";
+	non-removable;
+	disable-wp; /* We don't have a write-protect detection */
+	bus-width = <8>;
+	xlnx,mio_bank = <0>;
+};
+
+/* SD1 */
+&sdhci1 {
+	status = "okay";
+	bus-width = <4>;
+	xlnx,mio_bank = <1>;
+	disable-wp; /* We don't have a write-protect detection */
+};
+
+&serdes {
+	status = "okay";
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&uart1 {
+	status = "okay";
+};
+
+&watchdog0 {
+	status = "okay";
+	reset-on-timeout;
+	timeout-sec = <5>;
+};
+
+&xilinx_ams {
+	status = "okay";
+};
+
+&ams_ps {
+	status = "okay";
+};
+
+&ams_pl {
+	status = "okay";
+};
+
+&gem2 {
+	status = "okay";
+	phy-handle = <&miamiphy>;
+	phy-mode = "sgmii";
+	nvmem-cells = <&gem2_mac>;
+	nvmem-cell-names = "mac-address";
+	phys = <&lane2 PHY_TYPE_SGMII 2 1 125000000>; /* Lane 2 refclk 1 */
+	/* Need 25MHz and 125MHz clocks (Xilinx drivers lack proper clk support) */
+	assigned-clocks      = <&si5345 0 5>, <&si5345 0 2>;
+	assigned-clock-rates =    <25000000>,   <125000000>;
+
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		miamiphy: ethernet-phy@1 {
+			/* PHY on the board */
+			device_type = "ethernet-phy";
+			reg = <0x1>;
+			reset-gpios = <&gpioex 2 1>; /* Active low, on port 2 of GPIO expander */
+		};
+	};
+};
diff --git a/arch/arm64/configs/topic_miamimp_defconfig b/arch/arm64/configs/topic_miamimp_defconfig
new file mode 100644
index 000000000000..66d293c764ad
--- /dev/null
+++ b/arch/arm64/configs/topic_miamimp_defconfig
@@ -0,0 +1,399 @@
+CONFIG_LOCALVERSION="-xilinx-v2018.1"
+CONFIG_SYSVIPC=y
+CONFIG_POSIX_MQUEUE=y
+CONFIG_AUDIT=y
+CONFIG_NO_HZ=y
+CONFIG_HIGH_RES_TIMERS=y
+CONFIG_BSD_PROCESS_ACCT=y
+CONFIG_TASKSTATS=y
+CONFIG_TASK_DELAY_ACCT=y
+CONFIG_TASK_XACCT=y
+CONFIG_TASK_IO_ACCOUNTING=y
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
+CONFIG_LOG_BUF_SHIFT=16
+CONFIG_CGROUPS=y
+CONFIG_BLK_DEV_INITRD=y
+# CONFIG_RD_BZIP2 is not set
+# CONFIG_RD_LZMA is not set
+CONFIG_EMBEDDED=y
+# CONFIG_COMPAT_BRK is not set
+CONFIG_SLAB=y
+CONFIG_PROFILING=y
+CONFIG_MODULES=y
+CONFIG_MODULE_UNLOAD=y
+CONFIG_ARCH_ZYNQMP=y
+CONFIG_NR_CPUS=8
+CONFIG_DEFAULT_MMAP_MIN_ADDR=32768
+CONFIG_TRANSPARENT_HUGEPAGE=y
+CONFIG_TRANSPARENT_HUGEPAGE_MADVISE=y
+CONFIG_CMA=y
+# CONFIG_DMI is not set
+CONFIG_COMPAT=y
+CONFIG_CPU_IDLE=y
+CONFIG_ARM_CPUIDLE=y
+CONFIG_CPU_FREQ=y
+CONFIG_CPU_FREQ_STAT=y
+CONFIG_CPU_FREQ_DEFAULT_GOV_ONDEMAND=y
+CONFIG_CPU_FREQ_GOV_USERSPACE=y
+CONFIG_CPUFREQ_DT=y
+CONFIG_NET=y
+CONFIG_PACKET=y
+CONFIG_UNIX=y
+CONFIG_XFRM_USER=y
+CONFIG_NET_KEY=y
+CONFIG_NET_KEY_MIGRATE=y
+CONFIG_INET=y
+CONFIG_IP_MULTICAST=y
+CONFIG_IP_PNP=y
+CONFIG_IP_PNP_DHCP=y
+CONFIG_IP_PNP_BOOTP=y
+CONFIG_IP_PNP_RARP=y
+CONFIG_SYN_COOKIES=y
+CONFIG_NETWORK_SECMARK=y
+CONFIG_NETFILTER=y
+CONFIG_NETFILTER_NETLINK_LOG=y
+CONFIG_NF_CONNTRACK=m
+CONFIG_NF_CT_NETLINK=m
+CONFIG_NETFILTER_XT_MARK=y
+CONFIG_NETFILTER_XT_TARGET_CHECKSUM=y
+CONFIG_NETFILTER_XT_TARGET_LOG=y
+CONFIG_NETFILTER_XT_TARGET_REDIRECT=m
+CONFIG_NETFILTER_XT_MATCH_CONNMARK=m
+CONFIG_NETFILTER_XT_MATCH_CONNTRACK=m
+CONFIG_NETFILTER_XT_MATCH_LIMIT=y
+CONFIG_NETFILTER_XT_MATCH_MAC=y
+CONFIG_NETFILTER_XT_MATCH_MULTIPORT=y
+CONFIG_NETFILTER_XT_MATCH_STATE=m
+CONFIG_NF_CONNTRACK_IPV4=m
+CONFIG_NF_NAT_IPV4=m
+CONFIG_IP_NF_IPTABLES=y
+CONFIG_IP_NF_FILTER=y
+CONFIG_IP_NF_TARGET_REJECT=y
+CONFIG_IP_NF_MANGLE=y
+CONFIG_NF_CONNTRACK_IPV6=m
+CONFIG_NF_NAT_IPV6=m
+CONFIG_IP6_NF_IPTABLES=y
+CONFIG_IP6_NF_FILTER=y
+CONFIG_IP6_NF_TARGET_REJECT=y
+CONFIG_IP6_NF_MANGLE=y
+CONFIG_BRIDGE_NF_EBTABLES=y
+CONFIG_BRIDGE_EBT_T_FILTER=y
+CONFIG_BRIDGE_EBT_T_NAT=y
+CONFIG_BRIDGE_EBT_MARK_T=y
+CONFIG_BRIDGE=y
+CONFIG_DNS_RESOLVER=y
+CONFIG_NET_PKTGEN=y
+CONFIG_CAN=m
+CONFIG_CAN_XILINXCAN=m
+CONFIG_BT=m
+CONFIG_BT_RFCOMM=y
+CONFIG_BT_RFCOMM_TTY=y
+CONFIG_BT_BNEP=y
+CONFIG_BT_BNEP_MC_FILTER=y
+CONFIG_BT_BNEP_PROTO_FILTER=y
+CONFIG_BT_HIDP=y
+CONFIG_BT_LEDS=y
+CONFIG_BT_HCIBTUSB=m
+CONFIG_BT_HCIBTSDIO=m
+CONFIG_BT_HCIUART=m
+CONFIG_BT_HCIUART_BCSP=y
+CONFIG_BT_HCIUART_ATH3K=y
+CONFIG_BT_HCIUART_LL=y
+CONFIG_BT_HCIUART_3WIRE=y
+CONFIG_BT_HCIUART_INTEL=y
+CONFIG_BT_HCIUART_QCA=y
+CONFIG_BT_HCIBCM203X=m
+CONFIG_BT_HCIBPA10X=m
+CONFIG_BT_HCIBFUSB=m
+CONFIG_BT_HCIVHCI=m
+CONFIG_BT_MRVL=m
+CONFIG_BT_MRVL_SDIO=m
+CONFIG_BT_ATH3K=m
+CONFIG_BT_WILINK=m
+CONFIG_CFG80211=m
+CONFIG_NL80211_TESTMODE=y
+CONFIG_CFG80211_CERTIFICATION_ONUS=y
+CONFIG_CFG80211_REG_CELLULAR_HINTS=y
+CONFIG_CFG80211_REG_RELAX_NO_IR=y
+CONFIG_CFG80211_WEXT=y
+CONFIG_MAC80211=m
+CONFIG_MAC80211_LEDS=y
+CONFIG_MAC80211_MESSAGE_TRACING=y
+CONFIG_MAC80211_DEBUG_MENU=y
+CONFIG_RFKILL=y
+CONFIG_RFKILL_INPUT=y
+CONFIG_RFKILL_GPIO=y
+CONFIG_UEVENT_HELPER_PATH="/sbin/hotplug"
+CONFIG_DEVTMPFS=y
+CONFIG_DEVTMPFS_MOUNT=y
+CONFIG_DMA_CMA=y
+CONFIG_CMA_SIZE_MBYTES=256
+CONFIG_ARM_CCI400_PMU=y
+CONFIG_CONNECTOR=y
+CONFIG_MTD=y
+CONFIG_MTD_CMDLINE_PARTS=y
+CONFIG_MTD_BLOCK=y
+CONFIG_MTD_CFI=y
+CONFIG_MTD_CFI_INTELEXT=y
+CONFIG_MTD_M25P80=y
+CONFIG_MTD_SPI_NOR=y
+# CONFIG_MTD_SPI_NOR_USE_4K_SECTORS is not set
+CONFIG_MTD_UBI=y
+CONFIG_MTD_UBI_FASTMAP=y
+CONFIG_MTD_UBI_BLOCK=y
+CONFIG_OF_OVERLAY=y
+CONFIG_OF_CONFIGFS=y
+CONFIG_BLK_DEV_LOOP=y
+CONFIG_BLK_DEV_LOOP_MIN_COUNT=4
+CONFIG_BLK_DEV_RAM=y
+CONFIG_BLK_DEV_RAM_COUNT=4
+CONFIG_BLK_DEV_RAM_SIZE=65536
+CONFIG_XILINX_JESD204B=m
+CONFIG_XILINX_JESD204B_PHY=m
+CONFIG_EEPROM_AT24=y
+CONFIG_EEPROM_AT25=m
+CONFIG_TI_ST=y
+CONFIG_BLK_DEV_SD=y
+CONFIG_BLK_DEV_SR=m
+CONFIG_ATA=y
+CONFIG_SATA_AHCI_PLATFORM=y
+CONFIG_AHCI_CEVA=m
+# CONFIG_ATA_SFF is not set
+CONFIG_NETDEVICES=y
+CONFIG_TUN=y
+# CONFIG_NET_VENDOR_ALACRITECH is not set
+# CONFIG_NET_VENDOR_AMAZON is not set
+# CONFIG_NET_VENDOR_AMD is not set
+# CONFIG_NET_VENDOR_AQUANTIA is not set
+# CONFIG_NET_VENDOR_ARC is not set
+CONFIG_MACB=m
+# CONFIG_NET_VENDOR_BROADCOM is not set
+# CONFIG_NET_VENDOR_EZCHIP is not set
+# CONFIG_NET_VENDOR_HISILICON is not set
+# CONFIG_NET_VENDOR_HUAWEI is not set
+# CONFIG_NET_VENDOR_INTEL is not set
+# CONFIG_NET_VENDOR_MARVELL is not set
+# CONFIG_NET_VENDOR_MELLANOX is not set
+# CONFIG_NET_VENDOR_MICREL is not set
+# CONFIG_NET_VENDOR_MICROCHIP is not set
+# CONFIG_NET_VENDOR_NATSEMI is not set
+# CONFIG_NET_VENDOR_NETRONOME is not set
+# CONFIG_NET_VENDOR_QUALCOMM is not set
+# CONFIG_NET_VENDOR_RENESAS is not set
+# CONFIG_NET_VENDOR_ROCKER is not set
+# CONFIG_NET_VENDOR_SAMSUNG is not set
+# CONFIG_NET_VENDOR_SEEQ is not set
+# CONFIG_NET_VENDOR_SOLARFLARE is not set
+# CONFIG_NET_VENDOR_SMSC is not set
+# CONFIG_NET_VENDOR_STMICRO is not set
+# CONFIG_NET_VENDOR_VIA is not set
+# CONFIG_NET_VENDOR_WIZNET is not set
+# CONFIG_NET_VENDOR_XILINX is not set
+# CONFIG_NET_VENDOR_SYNOPSYS is not set
+CONFIG_MICREL_PHY=y
+CONFIG_XILINX_GMII2RGMII=m
+CONFIG_USB_USBNET=y
+# CONFIG_WLAN_VENDOR_ADMTEK is not set
+# CONFIG_WLAN_VENDOR_ATH is not set
+# CONFIG_WLAN_VENDOR_ATMEL is not set
+CONFIG_BRCMFMAC=m
+# CONFIG_WLAN_VENDOR_CISCO is not set
+# CONFIG_WLAN_VENDOR_INTEL is not set
+# CONFIG_WLAN_VENDOR_INTERSIL is not set
+# CONFIG_WLAN_VENDOR_MARVELL is not set
+# CONFIG_WLAN_VENDOR_MEDIATEK is not set
+# CONFIG_WLAN_VENDOR_RALINK is not set
+CONFIG_RTL8192CU=m
+# CONFIG_RTLWIFI_DEBUG is not set
+# CONFIG_WLAN_VENDOR_RSI is not set
+# CONFIG_WLAN_VENDOR_ST is not set
+# CONFIG_WLAN_VENDOR_TI is not set
+# CONFIG_WLAN_VENDOR_ZYDAS is not set
+# CONFIG_WLAN_VENDOR_QUANTENNA is not set
+CONFIG_INPUT_EVDEV=y
+CONFIG_KEYBOARD_GPIO=y
+# CONFIG_MOUSE_PS2 is not set
+CONFIG_VT_HW_CONSOLE_BINDING=y
+CONFIG_SERIAL_8250=m
+CONFIG_SERIAL_8250_RUNTIME_UARTS=2
+CONFIG_SERIAL_OF_PLATFORM=m
+CONFIG_SERIAL_UARTLITE=m
+CONFIG_SERIAL_XILINX_PS_UART=y
+CONFIG_SERIAL_XILINX_PS_UART_CONSOLE=y
+CONFIG_SERIAL_DEV_BUS=y
+CONFIG_SERIAL_DEV_CTRL_TTYPORT=y
+# CONFIG_HW_RANDOM is not set
+CONFIG_I2C_CHARDEV=y
+CONFIG_I2C_MUX=y
+CONFIG_I2C_MUX_PCA9541=y
+CONFIG_I2C_MUX_PCA954x=y
+CONFIG_I2C_CADENCE=y
+CONFIG_I2C_XILINX=m
+CONFIG_SPI=y
+CONFIG_SPI_BITBANG=y
+CONFIG_SPI_CADENCE=m
+CONFIG_SPI_ZYNQMP_GQSPI=y
+CONFIG_PTP_1588_CLOCK=y
+CONFIG_GPIOLIB=y
+CONFIG_GPIO_SYSFS=y
+CONFIG_GPIO_XILINX=y
+CONFIG_GPIO_ZYNQ=y
+CONFIG_GPIO_PCA953X=y
+CONFIG_GPIO_PCA953X_IRQ=y
+CONFIG_GPIO_PCF857X=y
+CONFIG_POWER_RESET_GPIO=y
+CONFIG_POWER_RESET_LTC2952=y
+CONFIG_CHARGER_GPIO=y
+CONFIG_BATTERY_GAUGE_LTC2941=m
+CONFIG_SENSORS_IIO_HWMON=y
+CONFIG_PMBUS=y
+CONFIG_SENSORS_MAX20751=y
+CONFIG_WATCHDOG=y
+CONFIG_CADENCE_WATCHDOG=y
+CONFIG_MFD_TPS65086=y
+CONFIG_REGULATOR=y
+CONFIG_REGULATOR_FIXED_VOLTAGE=y
+CONFIG_REGULATOR_GPIO=y
+# CONFIG_RC_CORE is not set
+CONFIG_DRM=y
+CONFIG_DRM_ADI_AXI_HDMI=m
+CONFIG_DRM_PANEL_SIMPLE=y
+CONFIG_DRM_I2C_ADV7511=y
+CONFIG_DRM_I2C_ADV7511_AUDIO=y
+# CONFIG_DRM_I2C_ADV7533 is not set
+CONFIG_BACKLIGHT_LCD_SUPPORT=y
+CONFIG_BACKLIGHT_CLASS_DEVICE=y
+CONFIG_SOUND=y
+CONFIG_SND=y
+# CONFIG_SND_DRIVERS is not set
+# CONFIG_SND_SPI is not set
+CONFIG_SND_USB_AUDIO=m
+CONFIG_SND_SOC=y
+CONFIG_SND_SOC_XILINX_DP=m
+CONFIG_USB=y
+CONFIG_USB_ANNOUNCE_NEW_DEVICES=y
+# CONFIG_USB_DEFAULT_PERSIST is not set
+CONFIG_USB_OTG=y
+CONFIG_USB_OTG_FSM=y
+CONFIG_USB_XHCI_HCD=y
+CONFIG_USB_STORAGE=y
+CONFIG_USB_UAS=y
+CONFIG_USB_DWC3=y
+CONFIG_USB_DWC3_ULPI=y
+CONFIG_NOP_USB_XCEIV=y
+CONFIG_USB_ULPI=y
+CONFIG_USB_GADGET=y
+CONFIG_USB_GADGET_XILINX=y
+CONFIG_USB_CONFIGFS=y
+CONFIG_USB_CONFIGFS_EEM=y
+CONFIG_USB_CONFIGFS_MASS_STORAGE=y
+CONFIG_USB_ETH=m
+CONFIG_USB_ETH_EEM=y
+CONFIG_USB_MASS_STORAGE=m
+CONFIG_USB_ULPI_BUS=y
+CONFIG_MMC=y
+CONFIG_MMC_SDHCI=y
+CONFIG_MMC_SDHCI_PLTFM=y
+CONFIG_MMC_SDHCI_OF_ARASAN=y
+CONFIG_NEW_LEDS=y
+CONFIG_LEDS_CLASS=y
+CONFIG_LEDS_GPIO=y
+CONFIG_LEDS_TRIGGER_TIMER=y
+CONFIG_LEDS_TRIGGER_ONESHOT=y
+CONFIG_LEDS_TRIGGER_DISK=y
+CONFIG_LEDS_TRIGGER_MTD=y
+CONFIG_LEDS_TRIGGER_HEARTBEAT=y
+CONFIG_LEDS_TRIGGER_BACKLIGHT=y
+CONFIG_LEDS_TRIGGER_CPU=y
+CONFIG_LEDS_TRIGGER_GPIO=y
+CONFIG_LEDS_TRIGGER_DEFAULT_ON=y
+CONFIG_LEDS_TRIGGER_TRANSIENT=y
+CONFIG_LEDS_TRIGGER_CAMERA=y
+CONFIG_LEDS_TRIGGER_PANIC=y
+CONFIG_EDAC=y
+CONFIG_EDAC_SYNOPSYS=y
+CONFIG_EDAC_ZYNQMP_OCM=y
+CONFIG_RTC_CLASS=y
+CONFIG_RTC_DRV_ZYNQMP=y
+CONFIG_DMADEVICES=y
+CONFIG_XILINX_DMA_ENGINES=y
+CONFIG_XILINX_DPDMA=m
+CONFIG_XILINX_FRMBUF=m
+CONFIG_XILINX_DMA=m
+CONFIG_XILINX_ZYNQMP_DMA=y
+CONFIG_DMATEST=m
+CONFIG_UIO=y
+CONFIG_UIO_PDRV_GENIRQ=m
+CONFIG_UIO_DMEM_GENIRQ=m
+CONFIG_UIO_XILINX_APM=m
+CONFIG_STAGING=y
+CONFIG_COMMON_CLK_XLNX_CLKWZRD=m
+CONFIG_XILINX_FCLK=y
+CONFIG_COMMON_CLK_SI514=y
+CONFIG_COMMON_CLK_SI544=y
+CONFIG_COMMON_CLK_SI570=y
+CONFIG_COMMON_CLK_CDCE925=y
+# CONFIG_COMMON_CLK_XGENE is not set
+CONFIG_COMMON_CLK_ZYNQMP=y
+CONFIG_ARM_SMMU=y
+CONFIG_REMOTEPROC=y
+CONFIG_ZYNQMP_R5_REMOTEPROC=m
+CONFIG_XILINX_VCU=m
+CONFIG_IIO=y
+CONFIG_AD799X=y
+CONFIG_XILINX_XADC=y
+CONFIG_XILINX_AMS=y
+CONFIG_RESET_CONTROLLER=y
+CONFIG_ZYNQMP_RESET_CONTROLLER=y
+CONFIG_PHY_XILINX_ZYNQMP=y
+CONFIG_RAS=y
+CONFIG_NVMEM_ZYNQMP=y
+CONFIG_FPGA=y
+CONFIG_ZYNQMP_FIRMWARE_DEBUG=y
+CONFIG_EXT4_FS=y
+CONFIG_EXT4_FS_POSIX_ACL=y
+CONFIG_EXT4_FS_SECURITY=y
+CONFIG_AUTOFS4_FS=y
+CONFIG_ISO9660_FS=m
+CONFIG_JOLIET=y
+CONFIG_UDF_FS=m
+CONFIG_MSDOS_FS=y
+CONFIG_VFAT_FS=y
+CONFIG_TMPFS=y
+CONFIG_TMPFS_POSIX_ACL=y
+CONFIG_HUGETLBFS=y
+CONFIG_JFFS2_FS=y
+CONFIG_JFFS2_SUMMARY=y
+CONFIG_JFFS2_FS_XATTR=y
+CONFIG_JFFS2_COMPRESSION_OPTIONS=y
+CONFIG_JFFS2_LZO=y
+CONFIG_JFFS2_RUBIN=y
+CONFIG_UBIFS_FS=m
+CONFIG_UBIFS_FS_ADVANCED_COMPR=y
+CONFIG_UBIFS_FS_ENCRYPTION=y
+CONFIG_SQUASHFS=y
+CONFIG_SQUASHFS_FILE_DIRECT=y
+CONFIG_SQUASHFS_DECOMP_MULTI=y
+CONFIG_SQUASHFS_LZO=y
+CONFIG_SQUASHFS_XZ=y
+# CONFIG_NETWORK_FILESYSTEMS is not set
+CONFIG_NLS_CODEPAGE_437=y
+CONFIG_NLS_ISO8859_1=y
+CONFIG_DEBUG_INFO=y
+CONFIG_DEBUG_FS=y
+CONFIG_MAGIC_SYSRQ=y
+# CONFIG_SCHED_DEBUG is not set
+# CONFIG_FTRACE is not set
+CONFIG_CRYPTO_ECDH=y
+# CONFIG_CRYPTO_MANAGER_DISABLE_TESTS is not set
+CONFIG_CRYPTO_CCM=y
+CONFIG_CRYPTO_GCM=y
+CONFIG_CRYPTO_CBC=y
+CONFIG_CRYPTO_ECB=y
+CONFIG_CRYPTO_CMAC=y
+CONFIG_CRYPTO_CRCT10DIF=y
+CONFIG_CRYPTO_MD5=y
+CONFIG_CRYPTO_ARC4=y
+# CONFIG_CRYPTO_HW is not set
+CONFIG_CRC7=y
-- 
2.17.1

